setup.py

```diff
@@ -42,6 +42,12 @@
         'torch>=1.3.0',
     ],
     extras_require={
+        'numpy': [
```

Comment:
IIUC this is not really an `extras` require because jax is imported in core funsor modules. I think for now it's fine to move this to a core dependency, but I'd like to make Funsor not rely on having all libraries installed. In particular, I've had some issues installing jax in the past, and I'd like to make sure we can develop Funsor with only one of JAX/PyTorch installed.

Reply:
Good point! I think we should have a simple solution here, let me think more about it...

Reply:
@fritzo It seems that [pypi does not support default extra_requires](https://github.com/pypa/setuptools/issues/1139) (I intended to make `pip install funsor` install `torch` by default and `pip install funsor[numpy]` skip `torch` and install `jax`). So my current solution is `pip install funsor[numpy]` will install both `torch` and `jax`. I hope it is fine for now.

Reply:
This looks fine for now, thanks for getting both numpy+jax backends to work!